File: src/main/scala/com/calclavia/graph/core/electric/NodeElectricJunction.scala
package com.calclavia.graph.core.electric

import com.calclavia.graph.core.electric.component.Junction
import nova.core.block.Block

/**
 * Wires are getNodes in the grid that will not have different terminals, but instead can connect omni-directionally.
 * Wires will be treated as junctions and collapsed.
 * @author Calclavia
 */
class NodeElectricJunction(parent: Block) extends com.calclavia.graph.api.energy.NodeElectricJunction(parent) with NodeAbstractElectric {

	override protected val block: Block = parent
	var junction: Junction = null

	override def current: Double = voltage * voltage / resistance

	override def toString: String = {
		"ElectricJunction [" + connections.size() + ", " + BigDecimal(voltage).setScale(2, BigDecimal.RoundingMode.HALF_UP) + "V]"
	}

	override def voltage: Double = junction.voltage
}

File: src/main/scala/com/calclavia/graph/api/energy/NodeElectricJunction.java
package com.calclavia.graph.api.energy;

import nova.core.block.Block;

/**
 * An electric node that acts as a junction (electrical wires) in an electric circuit.
 * Constructor requirement: Provider (An instance of {@link Block}
 * @author Calclavia
 */
public abstract class NodeElectricJunction extends NodeElectric {

	public NodeElectricJunction(Block provider) {
		super(provider);
	}

	@Override
	public final String getID() {
		return "electricJunction";
	}
}
